Packs/LogrhythmV2/Integrations/LogRhythmV2/LogRhythmV2.py

```diff
@@ -0,0 +1,2492 @@
+        headers = self._headers
+
+        response = self._http_request('GET', 'lr-admin-api/users', params=params, headers=headers)
+
+        return response
+
+    def lists_get_request(self, list_type, list_name, can_edit):
+        headers = self._headers
+
+        if list_type:
+            headers['listType'] = list_type
+
+        if list_name:
+            headers['name'] = list_name
+
+        if can_edit:
+            headers['canEdit'] = can_edit
+
+        response = self._http_request('GET', 'lr-admin-api/lists', headers=headers)
+
+        return response
+
+    def list_summary_create_update_request(self, list_type, name, enabled, use_patterns, replace_existing, read_access,
+                                           write_access, restricted_read, entity_name, need_to_notify, does_expire, owner):
+        data = {"autoImportOption": {"enabled": enabled, "replaceExisting": replace_existing, "usePatterns": use_patterns},
+                "doesExpire": does_expire, "entityName": entity_name, "owner": int(owner),
+                "listType": list_type, "name": name, "needToNotify": need_to_notify, "readAccess": read_access,
+                "restrictedRead": restricted_read, "writeAccess": write_access}
+        headers = self._headers
+
+        response = self._http_request('POST', 'lr-admin-api/lists', json_data=data, headers=headers)
+
+        return response
+
+    def list_details_and_items_get_request(self, list_id, max_items):
+        headers = self._headers
+
+        raw_response = self._http_request('GET', f'lr-admin-api/lists/{list_id}', headers=headers)
+        response = raw_response.copy()
+        if max_items:
+            items = response.get('items')[:int(max_items)]
+            response['items'] = items
+        return response, raw_response
+
+    def list_items_add_request(self, list_id, items):
+        if type(items) is dict:
+            items = [items]
+        data = {"items": items}
+        headers = self._headers
+        response = self._http_request('POST', f'lr-admin-api/lists/{list_id}/items', json_data=data, headers=headers)
+
+        return response
+
+    def list_items_remove_request(self, list_id, items):
+        if type(items) is dict:
+            items = [items]
+        data = {"items": items}
+        headers = self._headers
+
+        response = self._http_request('DELETE', f'lr-admin-api/lists/{list_id}/items', json_data=data, headers=headers)
+
+        return response
+
+    def execute_search_query_request(self, number_of_days, source_type, host_name, username, subject, sender,
+                                     recipient, hash_, url, process_name, object_, ipaddress, max_message,
+                                     query_timeout, entity_id):
+
+        headers = self._headers
+
+        # Create filter query
+        query = []
+
+        if host_name:
+            query.append(self.generate_query_item(23, 4, str(host_name)))
```

Comment:
what all those magic numbers mean?
maybe add an enum?

Reply:
I added a mapper with the filter type and value type